### ISSUE TITLE
Kobo ereader can sync, but won't download any books

### ISSUE THREAD

**Issue by bverwijst** [crocodilestick/Calibre-Web-Automated#66]
I've got an existing Calibre database, I've installed the automated version to the letter and I can see the full library. I also added the tool to convert to kepub files. Next thing I did was nuke my Kobo to start completely fresh (I don't care about stats or read books, I know what I've read).

Steps I've taken: 

- I enabled Sync with Kobo in the server settings
- I created a shelf with all books named Kobo
- I created a token and added the url to my Kobo config files

It syncs, I see all my books on the ereader, but they aren't downloaded. When I try it doesn't download the book it tries it, but nothing happens. When I click a book to open it, i get an error saying it can't download the book.

In order to test if this was a fault of the ereader or setup, I nuked my ereader setup once again and restarted my old Calibre Web + Calibre instance and that worked flawlessly when generating a new token setting up that url to sync.

Happy to help with more details if needed.

**Comment by jmarmstrong1207** [comment]
Could you post the Calibre web logs when you attempt to kobo sync?

**Comment by crocodilestick** [comment]
The latest updates to stock Calibre-Web seem to have borked Kobo sync for some users. Looking into what changes might be responsible

**Comment by bverwijst** [comment]
Here are the logs:

```[2024-09-04 21:41:23,355]  INFO {cps.shelf:347} Shelf Kobo Books created
[2024-09-04 21:44:46,430]  INFO {cps.kobo:1048} Init
[2024-09-04 21:44:46,864]  INFO {cps.kobo:146} Kobo library sync request received
[2024-09-04 21:44:47,900]  INFO {cps.kobo:146} Kobo library sync request received
[2024-09-04 21:45:05,105]  INFO {cps.server:320} webserver stop (restart=False)
[2024-09-04 21:45:05,105]  INFO {apscheduler.scheduler:212} Scheduler has been shut down
[2024-09-04 21:45:05,106]  INFO {cps.server:299} Performing shutdown of Calibre-Web
[2024-09-04 21:45:20,828]  WARN {cps.config_sql:384} Log path  not valid, falling back to default
[2024-09-04 21:45:20,861]  INFO {cps:166} *** "requests" version does not meet the requirements. Should: <2.32.0, Found: 2.32.3, please consider installing required version ***
[2024-09-04 21:45:20,861]  INFO {cps:175} Starting Calibre Web...
[2024-09-04 21:45:21,115]  WARN {py.warnings:109} /lsiopy/lib/python3.10/site-packages/flask_limiter/extension.py:337: UserWarning: Using the in-memory storage for tracking rate limits as no storage was explicitly specified. This is not recommended for production use. See: https://flask-limiter.readthedocs.io#configuring-a-storage-backend for documentation about configuring the storage backend.
  warnings.warn(
[2024-09-04 21:45:21,280]  INFO {apscheduler.scheduler:181} Scheduler started
[2024-09-04 21:45:21,280]  INFO {apscheduler.scheduler:895} Added job "delete temp" to job store "default"
[2024-09-04 21:45:21,281]  INFO {apscheduler.scheduler:895} Added job "end scheduled task" to job store "default"
[2024-09-04 21:45:21,281]  INFO {apscheduler.scheduler:895} Added job "immediately delete temp" to job store "default"
[2024-09-04 21:45:21,281]  INFO {apscheduler.executors.default:123} Running job "immediately delete temp (trigger: date[2024-09-04 21:45:21 BST], next run at: 2024-09-04 21:45:21 BST)" (scheduled at 2024-09-04 21:45:21.281176+01:00)
[2024-09-04 21:45:21,281]  INFO {apscheduler.scheduler:641} Removed job 964f918b50ab41b394042833cc84a3b7
[2024-09-04 21:45:21,284]  INFO {apscheduler.executors.default:144} Job "immediately delete temp (trigger: date[2024-09-04 21:45:21 BST], next run at: 2024-09-04 21:45:21 BST)" executed successfully
[2024-09-04 21:45:21,701]  WARN {py.warnings:109} /lsiopy/lib/python3.10/site-packages/pypdf/_crypt_providers/_cryptography.py:32: CryptographyDeprecationWarning: ARC4 has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.ARC4 and will be removed from this module in 48.0.0.
  from cryptography.hazmat.primitives.ciphers.algorithms import AES, ARC4
[2024-09-04 21:45:22,371]  INFO {cps.server:219} Starting Gevent server on [::]:8083
[2024-09-04 21:45:57,529]  INFO {cps.kobo:1048} Init
[2024-09-04 21:45:57,858]  INFO {cps.kobo:146} Kobo library sync request received
```

Some context, I deleted my old existing shelf and made a new one, restarted the server. The e-reader is completely factory reset with no books and the newly created shelf just contains a couple books to test. It gets the request, but does nothing with it it seems.

**Comment by crocodilestick** [comment]
Hopefully updating to the latest base CW version in the next update will solve this issue but if not we'll circle back

**Comment by jmarmstrong1207** [comment]
This is likely a setup issue because Kobo sync works for me. if you have a reverse proxy, heck https://github.com/janeczku/calibre-web/wiki/Setup-Reverse-Proxy because if not set up correctly, it won't download correctly as I issues with downloading too

**Comment by bverwijst** [comment]
I thought so too, but I've done it 100% local bypassing my reverse proxy using my local IP to sync and still have the issues that it won't sync. I'll wait until the next release and see if that changes anything.

**Comment by iFedix** [comment]
any update on this? same situation, i see my books in the kobo read but they need to be downloaded but if I click the download button nothing happens basically...

**Comment by crocodilestick** [comment]
@iFedix @bverwijst These issues should be now fixed in V2.1.0
